devito/types/tensor.py

```diff
@@ -31,16 +67,22 @@ class TensorFunction(AbstractTensor, Differentiable):
         Staggering of each component, needs to have the size of the tensor. Defaults
         to the Dimensions.
     """
+    _is_TimeDependent = False
     _sub_type = Function
-    _op_priority = Differentiable._op_priority + 2.
     _class_priority = 10
+    _op_priority = Differentiable._op_priority + 1.
```

Comment:
sorry for repeating myself, but the fact that you're accessing `Differentiable._op_priority` while _not_ subclassing it is symptomatic that there's something _conceptually_ off

Reply:
>while not subclassing it is symptomatic that there's something conceptually off

That's not true. This is purely a precedence concept, the same may a `mul` as precedence over an `add` but doesn't subclass it. There is nothing wrong with defining relative priorities between non-inherited object so that what to do when we have both is properly defined.